--- origsrc/kaffeine-0.8.7/kaffeine/configure.in.in	2008-01-20 11:50:52.000000000 -0600
+++ src/kaffeine-0.8.7/kaffeine/configure.in.in	2008-07-06 15:15:03.093750000 -0500
@@ -220,9 +220,15 @@
 dnl check for cdparanoia
 dnl --------------------
 
-KDE_CHECK_HEADER([cdda_interface.h], [with_cdparanoia=yes], [with_cdparanoia=no])
+AC_MSG_CHECKING([for cdparanoia])
+if "$PKG_CONFIG" --exists "libcdio_paranoia"; then
+	with_cdparanoia=yes
+	CFLAGS_PARANOIA=`"$PKG_CONFIG" --cflags libcdio_paranoia`
+	LIB_PARANOIA=`"$PKG_CONFIG" --libs libcdio_paranoia`
+	AC_MSG_RESULT([yes])
+else
+	AC_MSG_RESULT([no])
 
-if test "$with_cdparanoia" != "yes" ; then
 	echo ""
 	echo "-----------------------------------------"
 	echo "ERROR: Could not find cdparanoia headers."
@@ -231,6 +237,8 @@
 
 	AC_MSG_ERROR([could not find cdparanoia headers])
 fi
+AC_SUBST([CFLAGS_PARANOIA])
+AC_SUBST([LIB_PARANOIA])
 
 dnl ----------------------------
 dnl check for Ogg Vorbis encoder
--- origsrc/kaffeine-0.8.7/kaffeine/src/input/disc/Makefile.am	2007-06-03 08:55:05.000000000 -0500
+++ src/kaffeine-0.8.7/kaffeine/src/input/disc/Makefile.am	2008-07-06 15:15:03.109375000 -0500
@@ -13,6 +13,7 @@
 	-I$(top_srcdir)/kaffeine/src/ \
 	-I$(top_srcdir)/kaffeine/src/player-parts/kaffeine-part \
 	-I$(top_srcdir)/kaffeine/src/input/disc/plugins \
+	$(CFLAGS_PARANOIA) \
 	$(all_includes) 
 
 libkaffeinedisc_la_SOURCES = disc.cpp \
@@ -28,7 +29,7 @@
 libkaffeinedisc_la_LIBADD = $(top_builddir)/kaffeine/src/input/libkaffeineinput.la \
 	$(top_builddir)/kaffeine/src/player-parts/kaffeine-part/libkaffeinepart.la \
 	$(top_builddir)/kaffeine/src/input/disc/plugins/libkaffeineaudioencoder.la \
-	-lcdda_interface -lcdda_paranoia
+	$(LIB_PARANOIA)
 
 # this is where the XML-GUI resource file goes
 shellrcdir   = $(kde_datadir)/kaffeine
--- origsrc/kaffeine-0.8.7/kaffeine/src/input/disc/paranoia.cpp	2007-06-03 08:55:05.000000000 -0500
+++ src/kaffeine-0.8.7/kaffeine/src/input/disc/paranoia.cpp	2008-07-06 15:15:03.125000000 -0500
@@ -120,7 +120,7 @@
 {
 }
 
-void paranoiaCallback( long, int )
+void paranoiaCallback( long, paranoia_cb_mode_t )
 {
 }
 
@@ -396,7 +396,7 @@
 		curpos = currentSector;
 		endpos = endOfTrack;
 		if ( normalize ) {
-			len = CD_FRAMESIZE_RAW;
+			len = CDIO_CD_FRAMESIZE_RAW;
 			fn.open( IO_ReadWrite | IO_Truncate );
 			do {
 				buf = paranoia_read_limited( p, paranoiaCallback, 3 );
@@ -419,7 +419,7 @@
 			while ( curpos<endpos && len!=0 );
 
 			factor = 32767.0/max;
-			buf = new signed short[CD_FRAMESIZE_RAW];
+			buf = new signed short[CDIO_CD_FRAMESIZE_RAW];
 			fn.at( 0 );
 			f.open( IO_ReadWrite | IO_Truncate );
 			currentEncoder->start( encodingList[i].remove(0,3), encodingList[0], encodingList[1], encodingList[i].left(2) );
@@ -428,7 +428,7 @@
 				f.writeBlock( encoded, len );
 
 			do {
-				len = fn.readBlock( (char*)buf, CD_FRAMESIZE_RAW );
+				len = fn.readBlock( (char*)buf, CDIO_CD_FRAMESIZE_RAW );
 				if ( len>0 ) {
 					if ( max<32760 )
 						for ( n=0; n<len/2; ++n )
@@ -455,7 +455,7 @@
 			encoded = currentEncoder->getHeader( len );
 			if ( encoded )
 				f.writeBlock( encoded, len );
-			len = CD_FRAMESIZE_RAW;
+			len = CDIO_CD_FRAMESIZE_RAW;
 			do {
 				buf = paranoia_read_limited( p, paranoiaCallback, 3 );
 				if ( Q_BYTE_ORDER == Q_BIG_ENDIAN ) {
@@ -514,7 +514,7 @@
 	QString s, c;
 	long total;
 
-	total = CD_FRAMESIZE_RAW * (cdda_track_lastsector( d, t+1 )-cdda_track_firstsector( d, t+1 ) );
+	total = CDIO_CD_FRAMESIZE_RAW * (cdda_track_lastsector( d, t+1 )-cdda_track_firstsector( d, t+1 ) );
 	if ( total>(1048576 ) ) s = c.setNum(total/1048576.0, 'f', 2)+" "+i18n("MB");
 	else if ( total>1024 ) s = c.setNum(total/1024.0, 'f', 2)+" "+i18n("KB");
 	else s = c.setNum(total*1.0, 'f', 2)+" "+i18n("Bytes");
@@ -532,8 +532,8 @@
 	long total, time;
 	int m, s;
 
-	if ( t<0 ) total = CD_FRAMESIZE_RAW * (cdda_disc_lastsector( d )-cdda_disc_firstsector( d ) );
-	else total = CD_FRAMESIZE_RAW * (cdda_track_lastsector( d, t+1 )-cdda_track_firstsector( d, t+1 ) );
+	if ( t<0 ) total = CDIO_CD_FRAMESIZE_RAW * (cdda_disc_lastsector( d )-cdda_disc_firstsector( d ) );
+	else total = CDIO_CD_FRAMESIZE_RAW * (cdda_track_lastsector( d, t+1 )-cdda_track_firstsector( d, t+1 ) );
 	time = (8 * total) / (44100 * 2 * 16);
 	m = time/60;
 	s = time%60;
--- origsrc/kaffeine-0.8.7/kaffeine/src/input/disc/paranoia.h	2007-06-03 08:55:05.000000000 -0500
+++ src/kaffeine-0.8.7/kaffeine/src/input/disc/paranoia.h	2008-07-06 15:15:03.125000000 -0500
@@ -32,8 +32,7 @@
 
 extern "C"
 {
-#include <cdda_interface.h>
-#include <cdda_paranoia.h>
+#include <cdio/paranoia.h>
 }
 
 class KiloConfig : public ParanoiaSettings
@@ -92,7 +91,7 @@
 
 	long nTracks;
 	cdrom_drive *d;
-	cdrom_paranoia *p;
+	cdrom_paranoia_t *p;
 	long currentSector, endOfTrack;
 	bool isRunning;
 	QStringList encodingList;
