--- origsrc/kaffeine-0.8.7/kaffeine/src/instwizard.cpp	2007-06-03 08:55:14.000000000 -0500
+++ src/kaffeine-0.8.7/kaffeine/src/instwizard.cpp	2008-07-06 15:15:03.484375000 -0500
@@ -149,9 +156,9 @@
 	connect(&process, SIGNAL(receivedStdout(KProcess*,char*,int)), this, SLOT(slotStdout(KProcess*, char*, int)));
 	process.start(KProcess::Block, KProcess::Stdout);
 
-	if (stdout.contains('1'))
+	if (getStdout.contains('1'))
 		info << "<font color=\"DarkGreen\">" << i18n("Ok.") << "</font>";
-	else if (stdout.contains('0'))
+	else if (getStdout.contains('0'))
 		info << "<font color=\"DarkRed\">" << i18n("DMA mode off! For smooth DVD playback run as root:") +  " \"hdparm -d1 /dev/dvd\".</font>";
 	else
 		info << "<font color=\"DarkBlue\">" << i18n("Can't check DMA mode. Permission denied or no such device:")
@@ -274,5 +281,5 @@
 {
 	QString output = QString::fromLatin1(buffer, buflen);
 	kdDebug() << "WizardDialog: got from hdparm: " << output << "\n";
-	stdout.append(output);
+	getStdout.append(output);
 }
--- origsrc/kaffeine-0.8.7/kaffeine/src/instwizard.h	2007-06-03 08:55:14.000000000 -0500
+++ src/kaffeine-0.8.7/kaffeine/src/instwizard.h	2008-07-06 15:15:03.500000000 -0500
@@ -44,7 +44,7 @@
 
 	void internalWizard();
 
-	QString stdout;
+	QString getStdout;
 
 	InstWizard(const InstWizard &);
 	InstWizard operator=(const InstWizard &);
--- origsrc/kaffeine-0.8.7/kaffeine/src/player-parts/xine-part/kxinewidget.h	2007-11-15 03:05:57.000000000 -0600
+++ src/kaffeine-0.8.7/kaffeine/src/player-parts/xine-part/kxinewidget.h	2008-07-06 15:15:03.796875000 -0500
@@ -509,7 +509,7 @@
 	QTimer dvbOSDHideTimer;
 	xine_osd_t *dvbOSD;
 	QStringList dvbCurrentNext;
-	unsigned int dvbColor[256];
+	uint32_t dvbColor[256];
 	unsigned char dvbTrans[256];
 	void initDvbPalette();
 };
